client/state/test/initial-state.js

```diff
@@ -56,11 +62,60 @@ describe( 'initial-state', () => {
 			} );
 		} );
 		describe( 'persist-redux enabled', () => {
+			describe( 'switched user', () => {
+				describe( 'with recently persisted data and initial server data', () => {
+					var state,
+						savedState = {
+							postTypes: {
+								items: {
+									2916284: {
+										post: { name: 'post', label: 'Posts' },
+										page: { name: 'page', label: 'Pages' }
+									}
+								}
+							},
+							_timestamp: Date.now()
+						};
+					before( ( done ) => {
+						isReduxEnabled = true;
+						isSwitchedUser = true;
+						window.initialReduxState = { currentUser: { id: 123456789 } };
+						sinon.spy( console, 'error' );
+						sinon.stub( localforage, 'getItem' )
+							.returns(
+								new Promise( function( resolve ) {
+									resolve( savedState );
+								} )
+							);
+						const reduxReady = function( reduxStore ) {
+							state = reduxStore.getState();
+							done();
+						};
+						createReduxStoreFromPersistedInitialState( reduxReady );
+					} );
+					after( () => {
+						isReduxEnabled = false;
+						isSwitchedUser = false;
+						window.initialReduxState = null;
+						console.error.restore();
+						localforage.getItem.restore();
+					} );
+					it( 'builds store without errors', () => {
+						expect( console.error.called ).to.equal( false );
+					} );
+					it( 'does not build using local forage state', () => {
+						expect( state.postTypes.items[ 2916284 ] ).to.equal( undefined );
+					} );
+					it( 'does not add timestamp to store', () => {
+						expect( state._timestamp ).to.equal( undefined );
+					} );
+					it( 'builds state using server state', () => {
+						expect( state.currentUser.id ).to.equal( 123456789 );
```

Comment:
Is it possible to bypass this as well? This would likely be part of a separate PR. If the server state referred to here comes from data rendered into the page source - we'll want to ignore that. The server has no concept of whether a session is a support user or not - so it'll always render the original user's data. For example with the user bootstrap, we're ignoring the data rendered into the page and doing an XHR with the support token to fetch the user information.